src/robot/conf/settings.py

```diff
@@ -295,6 +298,13 @@ def _validate_flatten_keywords(self, values):
         except DataError as err:
             raise DataError("Invalid value for option '--flattenkeywords'. %s" % err)
 
+    def _validate_expandkeywords(self, values):
+        for opt in values:
+            if not (opt.lower().startswith(('name:', 'tag:'))):
+                raise DataError("Invalid value for option '--expandkeywords'."
+                                " Expected 'TAG:<pattern>', or "
+                                "'NAME:<pattern>' but got '%s'." % opt)            
```

Comment:
Good idea to have validation here.

Nitpicking:
- There are unnecessary parentheses in the 'if not` line.
- When using implicit string literal concatenation, it would be good to be consistent where the space is. Now the middle line both starts and ends with a space. First and second ending with a space would be a bit more clear.

Yeah, that's nitpicking for real. :joy:

Reply:
both fixed.